politeiawww/wsdcrdata.go

```diff
@@ -90,9 +100,73 @@ func (w *wsDcrdata) unsubFromAddr(address string) error {
 	return nil
 }
 
-// newWSDcrdata return a new wsDcrdata context.
-func newWSDcrdata() (*wsDcrdata, error) {
-	// Init websocket client
+// subToNewBlock subscribes to dcrdata events for new blocks
+func (w *wsDcrdata) subToNewBlock() error {
+	if w.isSubscribed(eventNewBlock) {
+		return errDuplicateSub
+	}
+	_, err := w.client.Subscribe(eventNewBlock)
+	if err != nil {
+		return fmt.Errorf("failed to subscribe: %v", err)
+	}
+	w.addSub(eventNewBlock)
+	log.Debugf("wsDcrdata subscribed to new block")
+	return nil
+}
+
+// reconnect stops the current client, initializes a new one, and subscribes
+// to each of the subscriptions that the previous client was subscribed to.
+func (w *wsDcrdata) reconnect(continueListening chan bool) {
+	previousSubscriptions := make(map[string]struct{})
+	for sub := range w.subscriptions {
+		previousSubscriptions[sub] = struct{}{}
+	}
+
+	w.client.Stop()
+	w.removeAllSubs()
+
+	go func() {
+		timeToWait := 1 * time.Minute
+		createdNewClient := false
+
+		for {
+			if !createdNewClient {
+				c, err := initDcrdataWSClient()
+				if err == nil {
+					w.client = c
+					createdNewClient = true
+					continueListening <- true
```

Comment:
I'm going to move this to after all the subscriptions are done. I'll make that change tonight.